Support FIFO queue required params

jhead opened the pull request:
Use of FIFO queues requires `aws-sdk` v2.7.4+ but these changes shouldn't break any backwards compatibility.

### Coverage
```
=============================== Coverage summary ===============================
Statements   : 95.56% ( 86/90 )
Branches     : 87.27% ( 48/55 )
Functions    : 100% ( 16/16 )
Lines        : 96.59% ( 85/88 )
================================================================================
```

robinjmurphy commented:
Thanks @jhead. Sorry for the delay with this. I'll try and take a look today for you.

robinjmurphy commented:
Are you able to get an example added to the `README` @jhead?

I think we probably have a bit of work to do to improve the documentation of this project but if we could get a simple FIFO example on the `README` that would be great.

jhead commented:
No problem! After some reading [here](http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/FIFO-queue-recommendations.html), I removed the validation that required both group ID and dedupe ID to be specified. If content-based dedupe is disabled on the queue and no dedupe ID is provided, the SDK will simply throw an error.

I simplified the tests I added a little and also added an example to the readme. If you'd like any further expansion on the tests, let me know. I wasn't 100% sure what would be sufficient there, but my changes are fairly minimal. The group ID and dedupe ID can be any string. The dedupe ID in practice is usually a SHA-256 (or other hash) of the message body, which is what AWS uses if you enable content-based dedupe.

I do have a suggestion. It would be handy if the producer could hold a default group ID specified in the constructor param and then pass that group ID down to every message, if the message itself doesn't have one specified. I think most people would only use a single group but they'd need to specify the group ID on every message object sent.

Example:
```Javascript
var producer = Producer.create({
  queueUrl: 'https://sqs.eu-west-1.amazonaws.com/account-id/queue-name',
  region: 'eu-west-1',
  groupId: 'myGroup1234'
});

// Entry sent with MessageGroupId set to 'myGroup1234'
producer.send('Hello world!');
```

The issue I ran into was that `entryFromObject()` does not have any context of the producer object.

robinjmurphy commented:
Thanks @jhead. Happy with these changes. Would you like me to merge this one and then we can look at the producer-level group configuration as a separate PR? Or would you prefer to work on it here?

jhead commented:
We can merge this one and track them separately 😄

robinjmurphy commented:
Thanks @jhead released as [1.5.0](https://github.com/bbc/sqs-producer/releases/tag/v1.5.0).
